backend/src/main/java/com/redhat/cloud/notifications/routers/internal/DuplicateNameMigrationResource.java

```diff
@@ -0,0 +1,147 @@
+@Path(API_INTERNAL + "/duplicate-name-migration")
+public class DuplicateNameMigrationResource {
+
+    final String ack = "i-am-sure-i-want-to-run-the-migration";
+
+    @Inject
+    EntityManager entityManager;
+
+    @PostConstruct
+    public void initialize() {
+        Log.infof("Duplicate name migration resource ack with value [%s]", ack);
+    }
+
+    @GET
+    @Produces(APPLICATION_JSON)
+    public DuplicateNameMigrationReport migrateDuplicateNames(@QueryParam("ack") String ack) {
+        if (!Objects.equals(this.ack, ack)) {
+            throw new BadRequestException(String.format("Invalid ack provided - should be [%s]", ack));
+        }
+
+        DuplicateNameMigrationReport migrationReport = new DuplicateNameMigrationReport();
+
+        Log.infof("Starting migration for duplicated names on behavior groups and integrations.");
+
+        while (true) {
+            DuplicateNameMigrationReport localReport = runMigrationStep();
+            migrationReport.updatedIntegrations += localReport.updatedIntegrations;
+            migrationReport.updatedBehaviorGroups += localReport.updatedBehaviorGroups;
+
+            if (localReport.updatedBehaviorGroups == 0 && localReport.updatedIntegrations == 0) {
+                break;
+            }
+        }
+
+        Log.infof("Finished updating duplicates names for behavior groups");
+
+        return migrationReport;
+    }
+
+    DuplicateNameMigrationReport runMigrationStep() {
+        DuplicateNameMigrationReport migrationReport = new DuplicateNameMigrationReport();
+
+        Log.infof("Running a migration step");
+
+        List<Object[]> repeatedIntegrationValues = entityManager.createNativeQuery(
+                        // string_agg concatenates all the grouped values with the delimiter (, in this case)
+                        "SELECT string_agg(CAST(id as character varying), ','), name FROM endpoints WHERE endpoint_type != :endpoint_type_email GROUP BY name, org_id HAVING count(*) > 1"
+                )
+                .setParameter("endpoint_type_email", EndpointType.EMAIL_SUBSCRIPTION.ordinal())
+                .getResultList();
+        Log.infof(
+                "Found %d different integration names across the  organizations that needs updating - each name could be used by multiple integrations",
+                repeatedIntegrationValues.size()
+        );
+
+        for (Object[] repeatedValue : repeatedIntegrationValues) {
+            migrationReport.updatedIntegrations += updateEndpoints(repeatedValue);
+        }
+
+        Log.infof("Finished step updating duplicates names for integrations");
+
+        List<Object[]> repeatedBehaviorGroupValues = entityManager.createNativeQuery(
+                // string_agg concatenates all the grouped values with the delimiter (, in this case)
+                "SELECT string_agg(CAST(id as character varying), ',') as ids, display_name FROM behavior_group GROUP BY display_name, org_id, bundle_id HAVING count(*) > 1"
```

Comment:
I wonder if the behavior groups should be ordered by creation date.

Reply:
That would be a nice improvement